code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm

```diff
@@ -0,0 +1,412 @@
+		owner.bleed(scab)
+		resetting_probability = 0
+	resetting_probability += (5*(current_cycle/10) * delta_time) // 10 iterations = >51% to itch
+	..()
+	return TRUE
+
+////Aiuri
+
+//impure
+/datum/reagent/impurity/aiuri
+	name = "Aivime"
+	description = "This reagent is known to interfere with the eyesight of a patient."
+	ph = 3.1
+	addiction_types = list(/datum/addiction/medicine = 1.5)
+	liver_damage = 0.1
+	//blurriness at the start of taking the med
+	var/cached_blurriness
+
+/datum/reagent/impurity/aiuri/on_mob_add(mob/living/owner, amount)
+	. = ..()
+	cached_blurriness = owner.eye_blurry
+	owner.set_blurriness(((creation_purity*10)*(volume/metabolization_rate)) + cached_blurriness)
+
+/datum/reagent/impurity/aiuri/on_mob_delete(mob/living/owner, amount)
+	. = ..()
+	if(owner.eye_blurry <= cached_blurriness)
+		return
+	owner.set_blurriness(cached_blurriness)
+
+////Hercuri
+
+//inverse
+/datum/reagent/inverse/hercuri
+	name = "Herignis"
+	description = "This reagent causes a dramatic raise in a patient's body temperature."
+	ph = 0.8
+	tox_damage = 0
+	addiction_types = list(/datum/addiction/medicine = 2.5)
+
+/datum/reagent/inverse/hercuri/on_mob_life(mob/living/carbon/owner, delta_time, times_fired)
+	. = ..()
+	var/heating = rand(creation_purity*10, creation_purity*30)
+	owner.reagents?.chem_temp += (heating * REM *normalise_creation_purity() * delta_time)
+	owner.adjust_bodytemperature(heating * (TEMPERATURE_DAMAGE_COEFFICIENT*REM), 50)
+	if(ishuman(owner))
+		var/mob/living/carbon/human/humi = owner
+		humi.adjust_coretemperature(heating * (TEMPERATURE_DAMAGE_COEFFICIENT*REM), 50)
+
+
+/datum/reagent/inverse/hercuri/expose_mob(mob/living/carbon/exposed_mob, methods=VAPOR, reac_volume)
+	. = ..()
+	if(!(methods & VAPOR))
+		return
+	exposed_mob.adjust_bodytemperature((reac_volume*creation_purity) * TEMPERATURE_DAMAGE_COEFFICIENT, 50)
+
+/datum/reagent/inverse/healing/tirimol
+	name = "Super Melatonin"//It's melatonin, but super!
+	description = "This will send the patient to sleep, adding a bonus to the efficacy of all reagents administered."
+	ph = 12.5 //sleeping is a basic need of all lifeforms
+	var/cached_reagent_list = list()
+	addiction_types = list(/datum/addiction/medicine = 5)
+
+//Makes patients fall asleep, then boosts the purirty of their medicine reagents if they're asleep
+/datum/reagent/inverse/healing/tirimol/on_mob_life(mob/living/carbon/owner, delta_time, times_fired)
+	switch(current_cycle)
+		if(1 to 10)//same delay as chloral hydrate
+			if(prob(50))
+				owner.emote("yawn")
+		if(10 to INFINITY)
+			owner.Sleeping(40)
+			. = 1
+			if(owner.IsSleeping() && !cached_reagent_list)
+				for(var/datum/reagent/reagent as anything in owner.reagents.reagent_list)
+					if(reagent in cached_reagent_list)
+						continue
+					if(istype(reagent, /datum/reagent/medicine))
+						if(!(reagent.creation_purity > reagent.inverse_chem_val))//Only affect pure types
+							continue
+						reagent.creation_purity *= 1.25
```

Comment:
what happen if the sleeping person with this chem in their system dies and then gets grinded up in a chef's gibber? will the reagents that end up in their meat end up having a greater than normal purity?

also, can you get your liver removed while you're asleep and this chem is in your system to keep the effects of its purity boost indefinitely?

Reply:
This is a good point. On_mob_metabolise() works regardless of liver, right? I should convert it to that.

Reply:
How do I test this? Injecting someone with it and removing their liver doesn't let me grind up and gain the injected reagents.